std/complex.d

```diff
@@ -142,15 +143,16 @@ struct Complex(T)  if (isFloatingPoint!T)
     }
 
     /// ditto
-    void toString(Char)(scope void delegate(const(Char)[]) sink,
+    void toString(Writer, Char)(scope Writer w,
                         FormatSpec!Char formatSpec) const
     {
         import std.math : signbit;
         import std.format : formatValue;
-        formatValue(sink, re, formatSpec);
-        if (signbit(im) == 0) sink("+");
-        formatValue(sink, im, formatSpec);
-        sink("i");
+        import std.array : put;
+        formatValue(w, re, formatSpec);
+        if (signbit(im) == 0) w("+");
```

Comment:
Yeah it should be `w.put(...)`. We need to stick to the output range API otherwise it will break when given a general output range.